tests/test_mcp_toolset.py

```
@@ -1125,3 +1128,193 @@ async def run_test():
+        custom_fields = ["tool_name", "tool_number"]
+        serialized = tool._serialize_context(context, fields=custom_fields)
+
+        assert "tool_name" in serialized
+        assert serialized["tool_name"] == "test_tool"
+        assert "tool_number" in serialized
+        assert serialized["tool_number"] == 5
+        assert "user_approved" not in serialized
+        assert "max_token_count" not in serialized
+        assert "llm" not in serialized
+
+    def test_build_dynamic_headers_with_context(self, suppress_migration_warnings):
```

Ruff (0.14.10)

Unused method argument: `suppress_migration_warnings`

(ARG002)